src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const add = (operand) => {
+    if (operand === undefined) {
+      const final = sum;
+
+      sum = 0;
+
+      return final;
+    }
+
+    sum += operand;
+
+    return add;
+  };
+
+  return add;
```

Comment:
you can return function right away
```suggestion
  return (operand) => {
    if (operand === undefined) {
      const final = sum;

      sum = 0;

      return final;
    }

    sum += operand;

    return add;
  };;
```